internal/controller/pod_webhook.go

```diff
@@ -45,11 +46,40 @@ func (a *PodAnnotator) Handle(ctx context.Context, req admission.Request) admiss
 	if err != nil {
 		return admission.Errored(400, fmt.Errorf("could not decode pod: %v", err))
 	}
-
+	log.FromContext(ctx).Info("webhook got pod ", "name", pod.Name)
 	if !hasMIGResource(pod) {
 		return admission.Allowed("No nvidia.com/mig-* resource found, skipping mutation.")
-	} else {
-		performQuotaArithmetic(pod, req)
+	}
+
+	performQuotaArithmetic(pod, req)
+
+	// Add extended resource to resource limits
+	if pod.Spec.Containers[0].Resources.Limits == nil {
```

Comment:
This and every other instance, why are we assuming it is going to be `Container[0]` all the time? We support only one container per pod, but does it also have to be the the 1st one?

Reply:
You are correct. It should not be the first one, but for this phase, we are assuming the first one (a strong assumption). There was also an issue open for init containers, which we do not support. One can argue whether Webhook should reject multiple container workloads. if such changes are desired we should create a separate issue for it.

Reply:
We need to document this well.

Reply:
May be this works: https://github.com/openshift/instaslice-operator/issues/25 if needed I can add it to the codebase

Reply:
https://github.com/openshift/instaslice-operator/issues/151